cmdstanpy/model.py

```diff
@@ -205,9 +205,9 @@ def __init__(
                 self._compiler_options.add_include_path(path)
 
             # try to detect models w/out parameters, needed for sampler
-            if not cmdstan_version_before(
-                2, 27
-            ):  # unknown end of version range
+            if not cmdstan_version_before(2, 27) and cmdstan_version_before(
+                2, 36
+            ):
```

Comment:
I would add parans around the not just to make things super easy to grok:

`if (not cmdstan_version_before(2, 27)) and cmdstan_version_befor(2,36):`